### PR TITLE
feat(web): add way of silencing role with volume

### PR DESCRIPTION
**Pull request by triptu**
https://100ms-docs-git-silence-role-js-100mslive.vercel.app/docs/javascript/v2/advanced-features/volume-control

### CODE REVIEW

File: docs/javascript/v2/advanced-features/volume-control.mdx [hunk 1]
```diff
@@ -16,4 +18,53 @@ hmsActions.setVolume(volume);
...
+  hmsActions.setVolume(0, peer.audioTrack);
+}
+```
+
+## Ideas
+
+### Proximity based audio for the Metaverse
+
+If you're building a 2D/3D world, where peers can navigate around and you want the audio to be based on how close two people are, you 
+can use this action to decide an appropriate audio level based on proximity.
```
Comment (saikatmitra91) [inline docs/javascript/v2/advanced-features/volume-control.mdx:51]: can use this action went to newline. can we fix this ?

File: docs/javascript/v2/advanced-features/volume-control.mdx [hunk 2]
```diff
@@ -16,4 +18,53 @@ hmsActions.setVolume(volume);
...
 hmsActions.setVolume(volume, "<track_id>");
+
+// get the currrent volume for a specific track
+hmsStore.getState(selectAudioTrackVolume("<track_id>"));
+```
+
+## Gotchas
+
+Availability of peer doesn't mean availability of their track id. The below code can silence the whole room if peer's audio
+track is not yet present.
```
Comment (saikatmitra91) [inline docs/javascript/v2/advanced-features/volume-control.mdx:29]: i think by mistake we added new line here

File: docs/javascript/v2/advanced-features/volume-control.mdx [hunk 3]
```diff
@@ -16,4 +18,53 @@ hmsActions.setVolume(volume);
...
+
+### Silence a role
+
+To silence a particular role, you can get all the peers for the role and silence the tracks in a loop - 
+
+```js
+function setVolumeByRole(roleName, volume) {
+  const peers = hmsStore.getState(selectPeersByRole(roleName));
+  for (const peer of peers) {
+    hmsActions.setVolume(volume, peer.audioTrack);
```
Comment (saikatmitra91) [inline docs/javascript/v2/advanced-features/volume-control.mdx:62]: can we add a check for `peer.audioTrack` here too ?

**Review by saikatmitra91 — CHANGES_REQUESTED** [review]

File: docs/javascript/v2/advanced-features/volume-control.mdx [hunk 4]
```diff
@@ -16,4 +22,76 @@ hmsActions.setVolume(volume);
...
+
+
+```js
+const peer = hmsStore.getState(selectPeerById("peer-id"));
+if (peer?.audioTrack) {
+  hmsActions.setVolume(0, peer.audioTrack);
+}
+```
+
+## Ideas
```
Comment (arjunattam) [inline docs/javascript/v2/advanced-features/volume-control.mdx:72]: love this section <3

Comment (triptu) [reply docs/javascript/v2/advanced-features/volume-control.mdx:72]: it's part of a few other pages as well - https://www.100ms.live/docs/javascript/v2/features/change-role#ideas
https://www.100ms.live/docs/javascript/v2/advanced-features/peer-metadata#ideas

**Comment by arjunattam** [comment]
ready to merge? @triptu

**Review by saikatmitra91 — APPROVED** [review]

**Pull request merged**